### ISSUE TITLE
wave pattern and automatization

### ISSUE THREAD

**Issue by lucasantoro** [Muskman/uwb-tools#1]
Hello,
How can i change the wave propagation pattern? for example in a circular pattern?
And, is there a method, to automatize the procedure, to compute the best location of anchor?

**Comment by Muskman** [comment]
We use the function isPtInUwbView2D to determine if the point is in the range of the sensor. The sensor wave propagation pattern used in our model is a quadrilateral given in the text and according to that we check through the function if a point lies in that quadrilateral or not. You could modify the function so as to work with a circular or any other pattern. For a circular pattern with detection range r, you have to check if sqrt(x^2+y^2) <= r

The problem of finding the best possible location is really hard and we are currently working on how to automate it.

**Comment by lucasantoro** [comment]
thank you for the response dear.
It is interesting this report, but, maybe, considering the fact, that for uwb trilateration in 2D/3D at least 3/4 anchors are needed, you can start with placing the anchors on the outer area, than placing the anchors in such way that each point on the map has covered at least from 4 anchors.
I don't know if genetic algorithm, or some adaptive method can be useful for this purpose, it's only my thought

**Comment by Muskman** [comment]
Thanks for your suggestions we are also thinking of adaptive methods although we are not sure how well they perform, we fear local search methods would get stuck at suboptimal solutions. The genetic algorithm is a good start, although we have to verify its stability. Let us know if you have any other techniques in mind.

**Comment by lucasantoro** [comment]
yes, i'm working on it now, if find some solution i will write to you. If you find some solution too, please write here
